File: app/src/main/java/com/openmobl/pttDriver/db/DriverDatabase.java
package com.openmobl.pttDriver.db;

import com.openmobl.pttDriver.model.Device;
import com.openmobl.pttDriver.model.Driver;

import java.util.List;

public interface DriverDatabase {
    void open();
    void close();

    List<Device> getDevices();
    List<Driver> getDrivers();

    Device getDevice(int id);
    Device getDevice(String macAddress);
    boolean deviceExists(int id);
    boolean deviceExists(String macAddress);
    void addDevice(Device device);
    void addOrUpdateDevice(Device device);
    void updateDevice(Device device);
    void removeDevice(int id);
    void removeDevice(Device device);

    Driver getDriver(int id);
    Driver getDriver(String name);
    boolean driverExists(int id);
    boolean driverExists(String name);
    void addDriver(Driver driver);
    void addOrUpdateDriver(Driver driver);
    void updateDriver(Driver driver);
    void removeDriver(int id);
    void removeDriver(Driver driver);
}

File: app/src/main/java/com/openmobl/pttDriver/db/DriverDatabase.kt
package com.openmobl.pttDriver.db

import com.openmobl.pttDriver.model.Device
import com.openmobl.pttDriver.model.Driver

interface DriverDatabase {
    fun open()
    fun close()
    val devices: List<Device>
    val drivers: List<Driver>
    fun getDevice(id: Int): Device?
    fun getDevice(macAddress: String): Device?
    fun deviceExists(id: Int): Boolean
    fun deviceExists(macAddress: String?): Boolean
    fun addDevice(device: Device)
    fun addOrUpdateDevice(device: Device)
    fun updateDevice(device: Device)
    fun removeDevice(id: Int)
    fun removeDevice(device: Device?)
    fun getDriver(id: Int): Driver?
    fun getDriver(name: String): Driver?
    fun driverExists(id: Int): Boolean
    fun driverExists(name: String?): Boolean
    fun addDriver(driver: Driver)
    fun addOrUpdateDriver(driver: Driver)
    fun updateDriver(driver: Driver)
    fun removeDriver(id: Int)
    fun removeDriver(driver: Driver?)
}